app/services/process_stripe_payment.rb

```diff
@@ -0,0 +1,25 @@
+class ProcessStripePayment
+  attr_reader :controller, :amount, :email, :token
+  
+  def initialize(controller, amount, email, token)
```

Comment:
I would prefer options hash to pass this many arguments. However I would not pass controller object - that is too much.